lib/controller.js

```diff
@@ -452,6 +455,165 @@ controller.connectToNetwork = function(opts) {
+            .then(function(currentSHA) {
+              // Figure out the human readable version for that code release
+              return updates.fetchBuildVersionforSHA(currentSHA);
+
+            });
+        });
+      }
+      // Otherwise, just resolve
+      else {
+        shaPromise = Promise.resolve();
+      }
+
+      // Once we have our Tessel and build version
+      return Promise.all([tesselPromise, shaPromise]).spread(function(tessel, currentVersion) {
+        // If it's not connected via USB, we can't update it
+        if (!tessel.usbConnection) {
+          return reject('Must have Tessel connected over USB to complete update. Aborting update.');
+        }
+
+        // If this Tessel isn't connected to the LAN
+        if (!tessel.lanConnection || tessel.lanConnection.authorized) {
+          // Warn the user that it's going to take a while
+          logs.warn('Warning: Could not detect Tessel over LAN. Updates will be much slower over USB.');
+        }
+
+        // If the update is forced or this build was requested
+        if (opts.build) {
+          // Fetch the requested build
+          return controller.updateTesselWithVersion(tessel, currentVersion, requestedVersion)
+            .then(resolve, reject);
+        } else {
+          // If they have requested the latest firmware
+          var verifiedVersion;
+          if (requestedVersion === 'latest') {
+            // Figure out what the version is
+            verifiedVersion = updates.fetchBuildList()
+              .then(function(builds) {
+                // Make sure we actually have builds
+                if (Object.keys(builds).length === 0) {
+                  return reject('Could not detect any builds on server.');
+                }
+                // Sort the builds by version
+                var buildArray = updates.buildDictToSortedArray(builds);
+
+                // Return the newest one
+                return buildArray[0].version;
+              });
+          }
+          // If they requested a specific version
+          else {
+            // Ensure that it exists by fetching the builds
+            verifiedVersion = updates.fetchSHAForBuildVersion(requestedVersion)
+              // If that function resolves, the sha exists so the version exists
+              .then(function() {
+                return requestedVersion;
+              })
+              // The version does not exist
+              .catch(reject);
+          }
+          // Once we have our verified version
+          verifiedVersion.then(function(requestedVersion) {
+            // Check if the current build is the same or newer if this isn't a forced update
+            if (!opts.force && currentVersion >= requestedVersion) {
+              // If it's not, close the Tessel connection and print the error message
+              var errMessage = tessel.name + ' is already on the latest firmware build. You can force an update with “tessel update --force”.';
```

Comment:
I think we should just type `tessel` for now since this will shortly be merged with the main `tessel` CLI. We have no other instances of mentioning `t2